client/asset/btc/btc_test.go

```diff
@@ -203,32 +172,65 @@ func (c *tRPCClient) GetBestBlockHeight() int64 {
 	return bestBlkHeight
 }
 
-func (c *tRPCClient) GetBestBlockHash() (*chainhash.Hash, error) {
-	c.blockchainMtx.RLock()
-	defer c.blockchainMtx.RUnlock()
-	var bestHash *chainhash.Hash
-	var bestBlkHeight int64
-	for height, hash := range c.mainchain {
-		if height >= bestBlkHeight {
-			bestBlkHeight = height
-			bestHash = hash
-		}
-	}
-	return bestHash, nil
-}
-
-func (c *tRPCClient) GetRawMempool() ([]*chainhash.Hash, error) {
-	return c.mempoolTxs, c.mpErr
-}
-
-func (c *tRPCClient) GetRawTransactionVerbose(txHash *chainhash.Hash) (*btcjson.TxRawResult, error) {
-	return c.mpVerboseTxs[txHash.String()], c.rawVerboseErr
-}
-
-func (c *tRPCClient) Disconnected() bool { return false }
-
-func (c *tRPCClient) RawRequest(method string, params []json.RawMessage) (json.RawMessage, error) {
+func (c *tRPCClient) RawRequest(_ context.Context, method string, params []json.RawMessage) (json.RawMessage, error) {
 	switch method {
+	case methodEstimateSmartFee:
+		optimalRate := float64(optimalFeeRate) * 1e-5 // ~0.00024
+		return json.Marshal(&btcjson.EstimateSmartFeeResult{
+			Blocks:  2,
+			FeeRate: &optimalRate,
+		})
+	case methodSendRawTransaction:
+		var tx *wire.MsgTx
+		_ = json.Unmarshal(params[0], &tx)
+		c.sentRawTx = tx
+		if c.sendErr == nil && c.sendHash == nil {
+			h := tx.TxHash()
+			return json.Marshal(&h)
+		}
+		if c.sendErr != nil {
+			return nil, c.sendErr
+		}
+		return json.Marshal(c.sendHash)
+	case methodGetTxOut:
+		if c.txOutErr != nil {
+			return nil, c.txOutErr
+		}
+		return json.Marshal(c.txOutRes)
+	case methodGetBlockHash:
+		var blockHeight int64
+		_ = json.Unmarshal(params[0], &blockHeight)
+		c.blockchainMtx.RLock()
+		defer c.blockchainMtx.RUnlock()
+		h, found := c.mainchain[blockHeight]
+		if !found {
+			return nil, fmt.Errorf("no test block at height %d", blockHeight)
+		}
+		return json.Marshal(h)
+	case methodGetBestBlockHash:
+		c.blockchainMtx.RLock()
+		defer c.blockchainMtx.RUnlock()
+		var bestHash *chainhash.Hash
+		var bestBlkHeight int64
+		for height, hash := range c.mainchain {
+			if height >= bestBlkHeight {
+				bestBlkHeight = height
+				bestHash = hash
+			}
+		}
+		return json.Marshal(bestHash)
+	case methodGetRawMempool:
+		if c.mpErr != nil {
+			return nil, c.mpErr
+		}
+		return json.Marshal(c.mempoolTxs)
```

Comment:
I can give you an illustrative diff in the morning, but one of the things that's giving you trouble is (un)marshalling into/from types that do not support it.  Note how most of the dcrjson types have hashes as strings, and the rpcclient methods that do return types like `[]*chainhash.Hash` will first have to unmarshal the json.RawMessage from the remote server into a string, then decode the hex.  Unfortunately that's the burden of mocking out a fake server.  We have to deal with that.